### DISCUSSION TITLE
cant finde sensor sensor.epex_spot_be_price

### DISCUSSION THREAD

**Discussion by Marown** [mampfes/ha_epex_spot#64] [Q&A]
Hi i am searching the sensor sensor.epex_spot_be_price
in your documentation there are posibillities to use the time with the best prices also to make a chart with those, but if i want to make those changes i cant find the variable sensor.epex_spot_be_price. Thanks for your help

**Comment by Marown** [comment]
I used sensor.epex_spot_data_lowest_price instead but it did not work

**Comment by mampfes** [comment]
Hi, after configuring the integration, you can find all available sensors in the following menu: Settings --> Devices & Services --> EPEX Spot:

![image](https://github.com/mampfes/ha_epex_spot/assets/375950/9d8702c8-2a15-4a64-b762-ab816c0d6f6d)

What do want to achieve?

**Reply by Marown** [reply]
Hi, i want to find the cheapest 3h in the day where i can charge my battery and heat some water etc. 
I found the sample code inside the documantation

```
`template:
  - sensor:
      - name: epex_start_low_period
        state: >-
          {% set ns = namespace(attr_dict=[]) %}
          {% for item in (state_attr('sensor.epex_spot_be_price', 'data'))[0:24] %}
              {%- set ns.attr_dict = ns.attr_dict + [(loop.index-1,item["price_eur_per_mwh"])] %}
          {% endfor %}
          {%- set price_map = dict(ns.attr_dict) %}
          {%- set price_sort = price_map.values()|list %}
          {%- set keys_list = price_map.keys()|list %}
          {%- set ns = namespace(combo=[]) %}
          {%- for p in keys_list %}
            {%- set p = p|int %}
            {%- if p < 22 %}
              {%- set ns.combo = ns.combo + [(p, ((price_sort)[p] + (price_sort)[p+1] + (price_sort)[p+2])|round(2))] %}
            {%- endif %}
          {%- endfor %}
          {%- set ns.combo = ns.combo[6:22] %}
          {%- set mapper = dict(ns.combo) %}
          {%- set key = mapper.keys()|list %}
          {%- set val = mapper.values()|list %}
          {%- set val_min = mapper.values()|min %}
          {{ key[val.index(val_min)]|string + ":00" }}`
```
          
The Sensor sensor.epex_spot_be_price is used but there is no sensore named lilke this.
Thanks

**Reply by mampfes** [reply]
I advise against using this method. Use "Lowest Price Sensor" or Service Call instead.

**Comment by mr-p666** [comment]
Check your entities. You should find something like 'at' or 'de' instead of 'be'.
As I understand, it depends on the source used.
However, with the latest versions you no longer need to use this code.  Check the documentation on the GitHub site with the current version 2.1.0.

**Reply by Marown** [reply]
Hi thanks, i actually looked into the service calls and i just can find the option to give me the x amount of hourse in one peace where it is the cheapest price. But i actually want the three cheapest hours on the whole 48h:
Also one question can i change the time zone ? becouse the tool takse the timestamp from the awattar api GMT +00 and i need +01
Thank you for your help

**Reply by Marown** [reply]
Also i want to trigger a helper in this cheap hours with the service
thanks

**Reply by mampfes** [reply]
The "Lowest Market Price" sensor ranks the hours of the current day. This can be also used for automations.

There is no need to change the time zone. It is just the internal representation that is in UTC, Home Assistant automatically takes care of the conversion into your local time zone.

**Reply by Marown** [reply]
Thanks for oyu rawnser, if i Use the Lowest Market Price sensor i just get the lowest hour on the day no rank if im not mystaking? 
If ido the Service call like so:

```
service: epex_spot.get_lowest_price_interval
data:
  earliest_start: "00:00:00"
  earliest_start_post: 1
  latest_end: "00:00:00"
  latest_end_post: 2
  duration: "03:00:00" 
```

it shows me the 3 cheapest hours in 48h, that is nice, but it would be ideal if the x houre you want would be given single.
Iguess i could use the Price sensor and filter it out my self in a way but i just wanted to know if there is something rdy, becouse this is very cool stuff here =)

**Reply by Marown** [reply]
And also if i do the Service call i get back the UTC 00 Time not the UTC 01 time

**Reply by mampfes** [reply]
71d04682059f077c080b3e96e2422a6d52acbf4c
